### PR TITLE
[PR-REVIEW] Fix KF version check bug

### PR DESCRIPTION
**Pull request by mnicely**
This PR fixes a bug found by @JasonAtNVIDIA.

If CuPy was not installed by source, say like `pip install cupy-cuda102` the version returned by `pkg_resources` didn't return **cupy**.

New version check
```python
# Check CuPy version
# Update to only check for v8.X in cuSignal 0.16
ver = cp.__version__
if (
    ver != "8.0.0b4"
    and ver != "8.0.0b5"
    and ver != "8.0.0rc1"
    and ver != "8.0.0"
):
    raise NotImplementedError(
         "Kalman Filter only compatible with CuPy v.8.0.0b4+"
    )
```

Per the comment, CuPy v8.0 should be officially release before cuSignal 0.16. In cuSignal 0.16, for KalmanFilter, minimal CuPy will be force to v.8.0.0.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by mnicely** [comment]
@awthomp @BradReesWork 
Something has changed in the CI that is causing all of the PRs to fail if you rerun the tests.
I've been able to recreate the errors with `branch-0.15` and local CI. 

The last one to pass all test is #150. 
Notice in the CI log if pulls `rapids-build-env-0.15.0a`. 
Where everything else is pulling `rapids-build-env-0.15.0a200719`

File: python/cusignal/estimation/filters.py [hunk 1]
```diff
@@ -200,8 +199,14 @@ def __init__(
...
-        ver = pkg_resources.get_distribution("cupy").version
-        if ver != "8.0.0b4" and ver != "8.0.0rc1" and ver != "8.0.0":
+        # Update to only check for v8.X in cuSignal 0.16
+        ver = cp.__version__
+        if (
+            ver != "8.0.0b4"
+            and ver != "8.0.0b5"
+            and ver != "8.0.0rc1"
+            and ver != "8.0.0"
+        ):
```
Comment (BradReesWork) [inline python/cusignal/estimation/filters.py:209]: Is it easier/better to define a set of valid versions up front and then do a
if ver in good_set: ....

File: python/cusignal/windows/windows.py [hunk 2]
```diff
@@ -302,8 +313,9 @@ def parzen(M, sym=True):
     na = cp.extract(n < -(M - 1) / 4.0, n)
     nb = cp.extract(abs(n) <= (M - 1) / 4.0, n)
     wa = 2 * (1 - abs(na) / (M / 2.0)) ** 3.0
-    wb = (1 - 6 * (abs(nb) / (M / 2.0)) ** 2.0 +
-          6 * (abs(nb) / (M / 2.0)) ** 3.0)
+    wb = (
+        1 - 6 * (abs(nb) / (M / 2.0)) ** 2.0 + 6 * (abs(nb) / (M / 2.0)) ** 3.0
```
Comment (BradReesWork) [inline python/cusignal/windows/windows.py:317]: Is it defined somewhere why you are using the number  '6'

File: python/cusignal/estimation/filters.py (hunk 1 above)
Comment (mnicely) [reply python/cusignal/estimation/filters.py:209]: Smart!

File: python/cusignal/windows/windows.py (hunk 2 above)
Comment (mnicely) [reply python/cusignal/windows/windows.py:317]: We'll need to ask @awthomp. This only got flagged because of formatting changes.

Comment (mnicely) [reply python/cusignal/windows/windows.py:317]: It's probably in the paper. Can I push this and then figure it out?

**Pull request merged**